### DISCUSSION TITLE
Clarification on persist middleware's partialize option

### DISCUSSION THREAD

**Discussion by anonimitoraf** [pmndrs/zustand#1273] [Q&A] [answered]
Hi, thank you for this library. It's been serving us well!

I just wanted to clarify the supposed behaviour of `partialize`. I have a store with some state like:
```typescript
{ a: SubStoreA, b: SubstoreB, c: SubstoreC }
```
I only want to persist `a`, so my `partialize` option looks something like: `store => _.pick(store, 'a')`. 

I've noticed that whenever `c`'s state changes (but `a` remains the same), zustand still persists my store's state.
Is this intended? If not, is there a way to only persist my store only when the substores picked in `partialize` change?

Happy to make a sample repo if it helps.

Thanks!

**Comment by dai-shi** [comment, marked as answer]
It's how it works now. `partialize` doesn't change subscription. It only change the shape of the state object before `serialize`. (I don't think it's easy to change the subscription from various aspects.) So, it's intended.

**Reply by dai-shi** [reply]
you could create your own `storage` to avoid duplicated writes.

**Reply by anonimitoraf** [reply]
I see. Thanks for the info. 
> you could create your own storage to avoid duplicated writes.

Sorry, what do you mean by this? Thanks!

**Reply by dai-shi** [reply]
https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md#getstorage
You can customize `storage`, instead of using the default `localStorage`.

Technically, you could solve your problem 👇 

> I've noticed that whenever `c`'s state changes (but `a` remains the same), zustand still persists my store's state.

**Reply by anonimitoraf** [reply]
Right, actually I already use a custom store. I guess what's missing is comparing the prev vs the curr state and only persisting if there's a diff?

**Reply by dai-shi** [reply]
So in your custom storage, you keep the prev value in a variable and check if there's a diff before writing to the real storage. I mean create custom `setItem` method. Oh, wait, we have `serialize` outside the storage (I confused with `jotai`). Sorry, this is too tricky, almost impossible.

**Reply by anonimitoraf** [reply]
Understood, I guess one option I have is to just split out the frequently-changing substore from the rest of the store?

Thanks

**Reply by dai-shi** [reply]
Yeah, I would do that.
